packages/camera/camera_android_camerax/test/android_camera_camerax_test.dart

```diff
@@ -561,6 +561,48 @@ void main() {
       verify(recording.close());
       verifyNoMoreInteractions(recording);
     });
+
+    test('stopVideoRecording throws a camera exception if '
+        'no recording is in progress', () async {
+      final AndroidCameraCameraX camera = AndroidCameraCameraX();
+      const String videoOutputPath = '/test/output/path';
+
+      camera.recording = null;
+      camera.videoOutputPath = videoOutputPath;
+
+      expect(() => camera.stopVideoRecording(0), throwsA(isA<CameraException>()));
+    });
+
+    test('stopVideoRecording throws a camera exception if '
+        'videoOutputPath is null', () async {
+      final AndroidCameraCameraX camera = AndroidCameraCameraX();
+      final MockRecording recording = MockRecording();
+
+      camera.recording = recording;
+      camera.videoOutputPath = null;
+
+      expect(() => camera.stopVideoRecording(0), throwsA(isA<CameraException>()));
+    });
+
+    test('calling stopVideoRecording twice stops the recording '
+        'and then throws a CameraException', () async {
+      final AndroidCameraCameraX camera = AndroidCameraCameraX();
+      final MockRecording recording = MockRecording();
+      final MockProcessCameraProvider processCameraProvider =
+      MockProcessCameraProvider();
+      final MockVideoCapture videoCapture = MockVideoCapture();
+      const String videoOutputPath = '/test/output/path';
+
+      camera.processCameraProvider = processCameraProvider;
+      camera.recording = recording;
+      camera.videoCapture = videoCapture;
+      camera.videoOutputPath = videoOutputPath;
+
+      final XFile file = await camera.stopVideoRecording(0);
+      assert(file.path == videoOutputPath);
```

Comment:
Should this be an expect statement instead of assert?

Reply:
Good catch! Changed

Reply:
It was actually in two places because I copied this test to make the test of "call start video recording twice", changed in both places now